framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -566,170 +556,100 @@ protected void reportPurityErrors(
-        FlowExpressionContext flowExprContext = null;
-        Set<PreOrPostcondition> postconditions = contractsUtils.getPostconditions(methodElement);
-
-        for (PreOrPostcondition p : postconditions) {
-            String expression = p.expression;
-            AnnotationMirror annotation = AnnotationUtils.fromName(elements, p.annotationString);
-
-            if (flowExprContext == null) {
-                flowExprContext =
-                        FlowExpressionContext.buildContextForMethodDeclaration(
-                                node, getCurrentPath(), checker.getContext());
-            }
-
-            // Only check if the postcondition concerns this checker
-            if (!atypeFactory.isSupportedQualifier(annotation)) {
-                continue;
-            }
-
             if (formalParamNames != null && formalParamNames.contains(expression)) {
+                @SuppressWarnings("CompilerMessages")
+                /*@CompilerMessageKey*/ String key =
+                        "contracts." + contract.kind.errorKey + ".expression.parameter.name";
                 checker.report(
                         Result.warning(
-                                "contracts.postcondition.expression.parameter.name",
+                                key,
                                 node.getName().toString(),
                                 expression,
                                 formalParamNames.indexOf(expression) + 1,
                                 expression),
                         node);
             }
 
-            try {
-                FlowExpressionParseUtil.parse(expression, flowExprContext, getCurrentPath(), false);
-            } catch (FlowExpressionParseException e) {
-                // report errors here
-                checker.report(e.getResult(), node);
-
-                // ignore expressions that do not parse
-                continue;
-            }
             checkFlowExprParameters(methodElement, expression);
         }
     }
 
-    /**
-     * Checks all conditional postcondition on the method {@code node} with element {@code
-     * methodElement}.
-     */
-    protected void checkConditionalPostconditions(
-            MethodTree node, ExecutableElement methodElement) {
-        FlowExpressionContext flowExprContext = null;
-        Set<ConditionalPostcondition> conditionalPostconditions =
-                contractsUtils.getConditionalPostconditions(methodElement);
-
-        for (ConditionalPostcondition p : conditionalPostconditions) {
-            String expression = p.expression;
-            boolean result = p.annoResult;
-            AnnotationMirror annotation = AnnotationUtils.fromName(elements, p.annotationString);
-
-            // Only check if the postcondition concerns this checker
-            if (!atypeFactory.isSupportedQualifier(annotation)) {
-                continue;
+    private void checkPostcondition(MethodTree node, AnnotationMirror annotation, Receiver expr) {
```

Comment:
A few of these methods were changed from protected to private and also lost their javadoc. What is the reason for this? At least some javadoc would be useful.